docker/Dockerfile.arm-unknown-linux-musleabi

```diff
@@ -26,10 +26,13 @@ RUN /musl-symlink.sh $CROSS_SYSROOT arm
 
 COPY qemu-runner base-runner.sh /
 
-ENV CARGO_TARGET_ARM_UNKNOWN_LINUX_MUSLEABI_LINKER=arm-linux-musleabi-gcc \
+COPY musl-gcc.sh /
+
+ENV CARGO_TARGET_ARM_UNKNOWN_LINUX_MUSLEABI_LINKER=musl-gcc.sh \
     CARGO_TARGET_ARM_UNKNOWN_LINUX_MUSLEABI_RUNNER="/qemu-runner arm" \
     CC_arm_unknown_linux_musleabi=arm-linux-musleabi-gcc \
     CXX_arm_unknown_linux_musleabi=arm-linux-musleabi-g++ \
     BINDGEN_EXTRA_CLANG_ARGS_arm_unknown_linux_musleabi="--sysroot=$CROSS_SYSROOT" \
     QEMU_LD_PREFIX="$CROSS_SYSROOT" \
-    RUST_TEST_THREADS=1
+    RUST_TEST_THREADS=1 \
+    CROSS_MUSL_LINKER=arm-linux-musleabi-gcc
```

Comment:
This can probably just be replaced with having something like `CROSS_TOOLCHAIN_PREFIX` and using that instead.